tests/components/LocationBias.test.tsx

```diff
@@ -0,0 +1,143 @@
+import { fireEvent, render, screen, waitFor } from '@testing-library/react';
+import { LocationBias } from '../../src/components/LocationBias';
+import { State, LocationBiasMethod } from '@yext/answers-headless-react';
+import * as locationOperations from '../../src/utils/location-operations';
+import { mockAnswersHooks, mockAnswersState, spyOnActions } from '../__utils__/mocks';
+import * as searchOperations from '../../src/utils/search-operations';
+
+jest.mock('@yext/answers-headless-react');
+
+const mockedStateVaDevice: Partial<State> = {
+  location: {
+    locationBias: {
+      latitude: 38.89552025579547,
+      longitude: -77.06991520330985,
+      displayName: 'Arlington, VA',
+      method: LocationBiasMethod.Device
+    }
+  }
+};
+
+const mockedStateNyIP: Partial<State> = {
+  location: {
+    locationBias: {
+      latitude: 40.741591687843005,
+      longitude: -74.00530254443494,
+      displayName: 'New York City, NY',
+      method: LocationBiasMethod.Ip
+    }
+  }
+};
+
+const mockedStateNoDisplayName: Partial<State> = {
+  location: {
+    locationBias: {
+      latitude: null,
+      longitude: null,
+      displayName: null,
+      method: null
+    }
+  },
+};
+
+const newGeoPosition = {
+  coords: {
+    accuracy: null,
+    altitude: null,
+    altitudeAccuracy: null,
+    heading: null,
+    latitude: 40.741591687843005,
+    longitude: -74.00530254443494,
+    speed: null,
+  },
+  timestamp: null
+};
+
+beforeEach(() => {
+  mockAnswersHooks({
+    mockedState: mockedStateVaDevice,
+    mockedActions: {
+      state: mockedStateVaDevice,
+      setUserLocation: jest.fn(),
+    }
+  });
+  jest.spyOn(locationOperations, 'getUserLocation').mockResolvedValue(newGeoPosition);
+  jest.spyOn(searchOperations, 'executeSearch').mockImplementation();
+});
+
+it('renders the proper text (location DisplayName, method, and update btn)', () => {
+  render(<LocationBias />);
+  const expectedLocationName = mockedStateVaDevice.location.locationBias.displayName;
+  const locationNameElement = screen.getByText(expectedLocationName);
+  expect(locationNameElement).toBeDefined();
+
+  const expectedMethodMessage = '(based on your device) -';
+  const basedOnMethodElement = screen.getByText(expectedMethodMessage);
+  expect(basedOnMethodElement).toBeDefined();
+
+  const updateLocationButton = screen.getByText('Update your location');
```

Comment:
I have a slight preference for accessing buttons using something like `screen.getByRole('button', {name: 'Update your location'})` so that we can be sure that it's a button and not just text being rendered

Reply:
changed all button accesses from getByText to getByRole('button' {name: ___ })